Libplanet/Action/Sys/SetValidator.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Numerics;
+using System.Text.Json.Serialization;
+using Bencodex.Types;
+using Libplanet.Consensus;
+using Libplanet.Crypto;
+
+namespace Libplanet.Action.Sys
+{
+    /// <summary>
+    /// A system action that promotes given address to the validator.
+    /// </summary>
+    [JsonConverter(typeof(SysActionJsonConverter))]
+    public sealed class SetValidator : IAction, IEquatable<SetValidator>, IEquatable<IAction>
+    {
+        /// <summary>
+        /// Creates a new instance of <see cref="SetValidator"/> action.
+        /// </summary>
+        /// <param name="validatorPublicKey">The public key of the validator operator
+        /// to be set.</param>
+        /// <param name="power">The power of the validator to be set.</param>
+        public SetValidator(PublicKey validatorPublicKey, BigInteger power)
+        {
+            ValidatorPublicKey = validatorPublicKey;
+            Power = power;
+        }
+
+        public SetValidator(Validator validator)
+            : this(validator.PublicKey, validator.Power)
+        {
+        }
```

Comment:
Seems a bit redundant. I think `SetValidator(Validator)` should suffice?

Reply:
For now, yes but in some cases such as mutation in headless, `SetValidator(PublicKey validatorPublicKey, BigInteger power)` can be used.